File: app/src/main/java/com/savedplaces/app/data/Place.kt
package com.savedplaces.app.data

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * يمثل مكاناً واحداً محفوظاً: اسمه، وإحداثياته، وتاريخ حفظه.
 */
@Entity(tableName = "places")
data class Place(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val name: String,
    val latitude: Double,
    val longitude: Double,
    val createdAt: Long = System.currentTimeMillis()
)

File: ios/Sources/Models/Place.swift
import Foundation
import SwiftData

/// يمثل مكاناً واحداً محفوظاً: اسمه، وإحداثياته، وتاريخ حفظه.
@Model
final class Place {
    var name: String
    var latitude: Double
    var longitude: Double
    var createdAt: Date

    init(name: String, latitude: Double, longitude: Double, createdAt: Date = Date()) {
        self.name = name
        self.latitude = latitude
        self.longitude = longitude
        self.createdAt = createdAt
    }
}
